lib/Runtime/Library/BuiltIn/BuiltIn.js

```diff
@@ -0,0 +1,60 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+"use strict";
+
+(function (intrinsic) {
+    var platform = intrinsic.BuiltIn;
```

Comment:
Unit test added.
Platform is a private variable sent in argument to this function that expose (for the moment) ONE private function called "registerFunction" which basically link a given method to a prototype.
Inside the given method, as the scope is now the scope of the given prototype (here, Array.prototype), we don't have access to the platform variable. That is why we use __chakraLibrary which is a global variable available only when the -LdChakraLib is set. This variable gives access to a set of private internal functions like toLength or toNumber that the consumers must not be able to exploit.

---
In reply to: [115388059](https://github.com/Microsoft/ChakraCore/pull/2898#discussion_r115388059) [](ancestors = 115388059)